spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-form-tracker.spec.js

```diff
@@ -390,6 +390,30 @@ describe('Google Analytics form tracking', function () {
     })
   })
 
+  describe('when tracking a form with text splitting enabled', function () {
+    beforeEach(function () {
+      element.setAttribute('data-ga4-form', JSON.stringify(attributes))
+      element.setAttribute('data-ga4-form-split-response-text', '')
+      expected = schema.mergeProperties(attributes, 'event_data')
+      expected.govuk_gem_version = 'aVersion'
+      expected.timestamp = '123456'
+      var tracker = new GOVUK.Modules.Ga4FormTracker(element)
+      tracker.init()
+    })
+
+    it('splits text into multiple dimensions', function () {
+      var overriddenText = JSON.parse(JSON.stringify(attributes))
+      var longText = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam pulvinar quam ac elit pulvinar, quis iaculis lorem tristique. Nunc hendrerit urna et metus iaculis lobortis. Aenean tempor, lectus at blandit aliquet, neque felis aliquet leo, ornare tristique nibh augue sit amet tellus. Curabitur sit amet maximus lorem. Nullam volutpat, dolor id mollis ultrices, dui libero venenatis nibh, vel mollis nulla diam vel leo. Vestibulum tempus quam sit amet nisi cursus, et cursus ipsum semper. Vivamus eu.'
+      var extraText = 'Extra characters after 500 characters.'
+      overriddenText.text = longText + extraText
+      element.setAttribute('data-ga4-form', JSON.stringify(overriddenText))
+      window.GOVUK.triggerEvent(element, 'submit')
+      expected.event_data.text = longText
+      expected.event_data.text_2 = extraText
+      expect(window.dataLayer[0]).toEqual(expected)
+    })
+  })
```

Comment:
Could there maybe be some more tests here? For example maybe:

- Exactly 500 characters
- A for loop to create a 2,500 character long string so that all 5 text params are tested? 
- Possibly boundaries (499 / 501 characters)
- Possibly below 500 chars (but I guess this is implied working by other tests)